x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java

```diff
@@ -381,9 +387,14 @@ private void fieldsMoved(LeafReaderContext ctx, int shard) throws IOException {
                 FieldWork field = fields[f];
                 rowStride[f] = field.rowStride(ctx);
                 storedFieldsSpec = storedFieldsSpec.merge(field.loader.rowStrideStoredFieldSpec());
+                SourceLoader sourceLoader = null;
```

Comment:
I think the code from here to the end of the loop can be moved outside and executed once.

```java
            if (false == storedFieldsSpec.equals(StoredFieldsSpec.NO_REQUIREMENTS)) {
                trackStoredFields(storedFieldsSpec, false);
                SourceLoader sourceLoader = null;
                if (storedFieldsSpec.requiresSource()) {
                    sourceLoader = shardContexts.get(shard).newSourceLoader.get();
                    storedFieldsSpec = storedFieldsSpec.merge(new StoredFieldsSpec(true, false, sourceLoader.requiredStoredFields()));
                }
                storedFields = new BlockLoaderStoredFieldsFromLeafLoader(
                    StoredFieldLoader.fromSpec(storedFieldsSpec).getLoader(ctx, null),
                    sourceLoader != null ? sourceLoader.leaf(ctx.reader(), null) : null
                );
            }
```

Reply:
good catch: 455ecd03a692119e10f244022f5c1d8f8559ae90